html/semantics/forms/constraints/form-validation-validity-stepMismatch.html

```diff
@@ -65,11 +65,14 @@
       tag: "input",
       types: ["number"],
       testData: [
-        {conditions: {step: "", value: "1"}, expected: false, name: "[target] The step attribute is not set"},
-        {conditions: {step: "", value: "-.8"}, expected: true, name: "[target] The step attribute is not set and the value attribute is a floating number"},
-        {conditions: {step: 2 * 1 * 1, value: ""}, expected: false, name: "[target] The value attribute is empty string"},
+        {conditions: {step: "", value: "1"}, expected: false, name: "[target] The step attribute is not set and the value attribute is an integer number"},
+        {conditions: {step: 2 * 1 * 1, value: ""}, expected: false, name: "[target] The value attibute is empty string"},
         {conditions: {step: 2 * 1 * 1, value: "2"}, expected: false, name: "[target] The value must match the step"},
-        {conditions: {step: 2 * 1 * 1, value: "3"}, expected: true, name: "[target] The value must mismatch the step"}
+        {conditions: {step: "0.003", value: "3.6"}, expected: false, name: "[target] No step mismatch when step is a floating number and value is its integral multiple"},
+        {conditions: {step: "1e-12", value: "-12345678.9"}, expected: false, name: "[target] No step mismatch when step is a floating number in exponent format and value is its integral multiple"},
+        {conditions: {step: "", value: "-.8"}, expected: true, name: "[target] The step attribute is not set and the value attribute is a floating number"},
+        {conditions: {step: 2 * 1 * 1, value: "3"}, expected: true, name: "[target] The value must mismatch the step"},
+        {conditions: {step: "3e-16", value: "17"}, expected: true, name: "[target] No step mismatch when step is a very small floating number and value is not its integral multiple"},
```

Comment:
Should this be "Step mismatch", instead of "No step mismatch"?

Reply:
This test also fails in Chrome it appears.

In general I'm having a hard time reviewing this because I can't tell which tests are new versus which are changes to existing tests. Could you revert all changes to existing tests, and maybe just add new ones at the end or something? You can do a separate PR to clean things up if needed.

Reply:
Thank you. Indeed, it should have been "Step mismatch", I've corrected it. I've also reverted the order (previously I sorted the tests by the `expected:` value) and added new tests to the end. Regarding failure in Chrome, I guess it's at the edge of the precisoin of the algorithm in Chrome, I upped the exponent by one will see whether that would work correctly in Chrome then.